File: ruby/grade-school/school.rb
class School

  attr_reader :name

  def initialize(name)
    @name = name
    @db   = Database.new
  end

  def db
    @db.to_hash
  end

  def add(student, grade_number)
    @db[grade_number] << student
  end

  def grade(grade_number)
    @db[grade_number]
  end

  def sort
    @db.sorted_keys.each_with_object({}) do |grade_number, sorted|
      sorted[grade_number] = grade(grade_number).sort
    end
  end

end

class Database

  def initialize
    @db = Hash.new { |hash, key| hash[key] = [] }
  end

  def []=(key, value)
    @db[key] = value
  end

  def [](key)
    @db[key]
  end

  def sorted_keys
    @db.keys.sort
  end

  def to_hash
    @db
  end

end

File: python/grade-school/school.py

class School:

  def __init__(self, name):
    self.name = name
    self.db = {}

  def add(self, student, grade_number):
    students_in_grade = self.grade(grade_number)
    students_in_grade.add(student)
    self.db[grade_number] = students_in_grade

  def grade(self, grade_number):
    return self.db.get(grade_number, set())

  def sort(self):
    sorted_db = {}
    for grade_number in sorted(self.db.keys()):
      sorted_db[grade_number] = tuple(sorted(self.db[grade_number]))
    return sorted_db